hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AUtils.java

```diff
@@ -820,66 +759,80 @@ private static AwsCredentialsProvider createAWSV2CredentialProvider(
       throw new IOException("Class " + credClass + " " + ABSTRACT_PROVIDER);
     }
     LOG.debug("Credential provider class is {}", className);
+    credentials =
+        getInstanceFromReflection(credClass, conf, uri, AwsCredentialsProvider.class, "create",
+            AWS_CREDENTIALS_PROVIDER);
+    return credentials;
+  }
+
+  /***
+   * Creates an instance of a class using reflection.
+   *
```

Comment:
May be worth listing here the sequence of constructors/methods that will be used (essentially what's in the code comments). It would help to clarify what kind of class this is useful for, I think.